### ISSUE TITLE
Range removal fails with specific bits, when runlength encoding was applied/removed

### ISSUE THREAD

**Issue by okrische** [RoaringBitmap/RoaringBitmap#67]
When addding these bits and then runOptimize and removing it, range removal does not empty the bitset. But i only found this combination, that does not work.

```java
	@Test
	public void testRangeRemovalWithEightBitsAndRunlengthEncoding() {
		final MutableRoaringBitmap bitmap = new MutableRoaringBitmap();
		bitmap.add(1); // index 1
		bitmap.add(2);
		bitmap.add(3);
		bitmap.add(4);
		bitmap.add(5);
		bitmap.add(7); // index 7

		bitmap.runOptimize();
		bitmap.removeRunCompression();
			
		// should remove from index 0 up to index 7 (excluding 8th)
		bitmap.remove(0, 8);
		assertTrue("fails with cardinality: "+bitmap.getCardinality(), bitmap.isEmpty());
	}
```

**Comment by lemire** [comment]
Should be fixed in release 0.5.18. Please verify.

**Comment by okrische** [comment]
Wow. Thanks!
